app/moves/middleware.js

```diff
@@ -11,28 +13,44 @@ const moveService = require('../../common/services/move')
 const { LOCATIONS_BATCH_SIZE } = require('../../config')
 
 const moveDateFormat = 'yyyy-MM-dd'
+const weekStartsOn = 1
 
-function makeMultipleRequests(service, moveDate, locationIdBatches) {
+function makeMultipleRequests(service, dateRange, locationIdBatches) {
   return Promise.all(
-    locationIdBatches.map(chunk => service({ moveDate, fromLocationId: chunk }))
+    locationIdBatches.map(chunk =>
+      service({ dateRange, fromLocationId: chunk })
+    )
   )
 }
 
+function getPeriod(date, interval) {
+  const method = interval >= 0 ? addDays : subDays
+  return format(method(parseISO(date), Math.abs(interval)), moveDateFormat)
+}
+
 module.exports = {
   redirectBaseUrl: (req, res) => {
     const today = format(new Date(), moveDateFormat)
     const currentLocation = get(req.session, 'currentLocation.id')
 
     if (currentLocation) {
-      return res.redirect(`${req.baseUrl}/${today}/${currentLocation}`)
+      return res.redirect(`${req.baseUrl}/day/${today}/${currentLocation}`)
     }
 
-    return res.redirect(`${req.baseUrl}/${today}`)
+    return res.redirect(`${req.baseUrl}/day/${today}`)
   },
   saveUrl: (req, res, next) => {
     req.session.movesUrl = req.originalUrl
     next()
   },
+  setPeriod: (req, res, next, period) => {
```

Comment:
Just had a small thought this morning - are this and status actually being used in the view right now?

If not I'd be tempted to remove the need for these.

Reply:
Ah, I can see its being used for some logic to display the header.

But looking at that logic in the template. If we replace `setMoveDate` with `setDateRange` in  the date param middleware call you could do the template logic based on length. If one item in the array, show first, if two, show the date range in the header?